File: algorithms-and-data-structures/algorithms-and-data-structures-1/problem-set-09/q04.java
	import java.util.Scanner;
	class Autor{
		private String nome;
		public Autor(String nome){
			this.nome=nome;
		}
		void print(){
			System.out.println(nome);
		}	
		
	}
	class Livro{
		private String titulo;
		private int anoPubli;
		private Autor ponteiro;
		public Livro(String titulo,int ano,String autor){
			this.titulo=titulo;
			anoPubli=ano;
			ponteiro=new Autor(autor);
			
		}
		void print(){
			System.out.println(titulo);
			System.out.println(anoPubli);
			ponteiro.print();
		}
		
	}
	public class q04{
		public static void main(String[] args){
			Scanner sc=new Scanner(System.in);
			String titulo=sc.nextLine();
			String autor=sc.nextLine();
			int anoPubli=sc.nextInt();
			Livro livro=new Livro(titulo,anoPubli,autor);
			livro.print();	
		}
	}

File: algorithms-and-data-structures/algorithms-and-data-structures-1/problem-set-09/q04.c
	#include <stdio.h>
	#include <string.h>
	#include <stdlib.h>
	struct Autor{
		char nome[255];	
	}typedef Autor;
	
	struct Livro{
		char titulo[255];
		int ano;
		Autor* ponteiro;	
	}typedef Livro;
	
	void readline(char* str,int tam){
		fgets(str,tam,stdin);
		str[strlen(str)-1]='\0';
	}
	
	void set(Livro* livro){
		char titulo[255],autor[255];
		int ano;
		readline(autor,255);
		readline(titulo,255);
		scanf("%d",&ano);
		(*livro).ponteiro=malloc(sizeof(Autor));
		strcpy((*((*livro).ponteiro)).nome,autor);
		strcpy((*livro).titulo,titulo);
		(*livro).ano=ano;	
		
	}
	
	void print(Livro* livro){
		printf("%s\n",(*((*livro).ponteiro)).nome);
		printf("%s\n",(*livro).titulo);
		printf("%d\n",(*livro).ano);
	}	
	
	int main(){
		Livro livro;
		set(&livro);
		print(&livro);
		
	}
